medicines/doc-index-updater/src/main.rs

```diff
@@ -91,6 +93,41 @@ fn create_redis_url(server: String, port: String, key: String) -> String {
     }
 }
 
+#[derive(serde::Serialize)]
+struct ErrorMessage {
+    code: u16,
+    message: String,
+}
+
+async fn handle_rejection(err: warp::Rejection) -> Result<impl warp::Reply, Infallible> {
+    let code;
+    let message;
+
+    if err.is_not_found() {
+        code = StatusCode::NOT_FOUND;
+        message = "NOT_FOUND";
+    } else if let Some(warp::reject::MethodNotAllowed { .. }) = err.find() {
+        code = StatusCode::METHOD_NOT_ALLOWED;
+        message = "METHOD_NOT_ALLOWED";
+    } else if let Some(warp::reject::UnsupportedMediaType { .. }) = err.find() {
+        code = StatusCode::UNSUPPORTED_MEDIA_TYPE;
+        message = "UNSUPPORTED_MEDIA_TYPE";
+    } else if let Some(AuthenticationFailed) = err.find() {
+        code = StatusCode::UNAUTHORIZED;
+        message = "AUTHENTICATION_FAILED";
+    } else {
+        code = StatusCode::INTERNAL_SERVER_ERROR;
+        message = "UNHANDLED_REJECTION";
+    }
+
+    let json = warp::reply::json(&ErrorMessage {
+        code: code.as_u16(),
+        message: message.into(),
+    });
+
+    Ok(warp::reply::with_status(json, code))
```

Comment:
Warp will automatically log the outgoing response, so we don't need to manually log here, right?